### PR TITLE
Android: Graphic setting description improvements

### PR DESCRIPTION
**Pull request by Ebola16**
Follow-up to #7591 and #7121.

Implements `SingleChoiceSettingDynamicDescriptions` and applies it to the Shader Compilation Mode setting.

Pinging @mahdihijazi @zackhow @MayImilae and @JosJuice in case you're interested in this.

screenshot_20190219-134448_dolphin emulator

### CODE REVIEW

**Comment by Ebola16** [comment]
Any review comments for this? This is the simplest PR I'm proposing.

File: Source/Android/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -229,7 +229,11 @@
...
+    <string name="shader_compilation_sync_uber_description">Synchronous (Ubershaders): Ubershaders will always be used. Provides a near stutter-free experience at the cost of high GPU performance requirements. Only recommended for high-end systems.</string>
+    <string name="shader_compilation_async_uber_description">Asynchronous (Ubershaders): Ubershaders will be used to prevent stuttering during shader compilation, but specialized shaders will be used when they will not cause stuttering. In the best case it eliminates shader compilation stuttering while having minimal performance impact, but results depend on video driver behavior.</string>
+    <string name="shader_compilation_async_skip_description">Asynchronous (Skip Drawing): Prevents shader compilation stuttering by not rendering waiting objects. Can work in scenarios where Ubershaders doesn\'t, at the cost of introducing visual glitches and broken effects. Not recommended, only use if the other options give poor results on your system.</string>
```
Comment (MayImilae) [inline Source/Android/app/src/main/res/values/strings.xml]: <pre>doesn\'t</pre>

I'm assuming that will be "doesn't" on devices, right?

Comment (Ebola16) [reply Source/Android/app/src/main/res/values/strings.xml]: Looks good on my device and apostrophes are done the same way elsewhere in strings.xml. 
Screenshot_20190320-200548_Dolphin Emulator

**Comment by Ebola16** [comment]
... Does anyone else want to review / merge this?

**Comment by Ebola16** [comment]
...nobody?

**Comment by jordan-woyak** [comment]
@Ebola16 Unfortunately I don't know if we have anyone right now that cares enough about Android to review or even test this.

**Comment by Dockolm** [comment]
The code looks good overall.
Though I'm not a fan of writing feature-specific code on SingleChoiceViewHolder. Can we consider to create another generic ViewHolder that is designed to take a description for each choice ?
Also, I'm still more or less a beginner at programming, so feel free to correct me if I'm wrong.

**Comment by Ebola16** [comment]
In `SettingsAdapter`, `TYPE_SINGLE_CHOICE` is tied to `SingleChoiceViewHolder`. If I create a new viewholder that displays the descriptions like this PR does, I think I'll need to create a new `SettingsItem.TYPE` and viewholder for each `SettingsItem.TYPE` if we want to expand this change to the other TYPEs. Since what I propose is less LOC, I like my approach better. I'm also relatively new to programming so let me know if I'm missing something.

**Comment by Dockolm** [comment]
Should be okay then, as long as we don't have many special cases like that.
Tested the GUI, works fine.
I can't test if the ubershader setting is applied correctly, my Android device can't run Dolphin. But it should be fine.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/viewholder/SingleChoiceViewHolder.java [hunk 2]
```diff
@@ -38,7 +38,34 @@ public void bind(SettingsItem item)
 
     if (item.getDescriptionId() > 0)
     {
-      mTextSettingDescription.setText(item.getDescriptionId());
+      // Special cases for SingleChoiceSetting options with long descriptions
```
Comment (leoetlino) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/viewholder/SingleChoiceViewHolder.java]: This doesn't really belong here. @mahdihijazi explained the issue with this approach in #7121: https://github.com/dolphin-emu/dolphin/pull/7121#discussion_r204168841

Comment (Ebola16) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/viewholder/SingleChoiceViewHolder.java]: I tried a different approach.

**Comment by Ebola16** [comment]
I tried a different approach but I'm not sure if I did it properly. Ready for review.

**Comment by Ebola16** [comment]
...any other reviews?

**Comment by Ebola16** [comment]
...if this looks good can it be merged? This is needed for future PRs.

**Comment by Helios747** [comment]
yolo

**Pull request merged**